runtime/compiler/runtime/J9CodeCacheManager.cpp

```diff
@@ -779,3 +803,26 @@ J9::CodeCacheManager::printOccupancyStats()
       codeCache->printOccupancyStats();
       }
    }
+
+
+int32_t
+J9::CodeCacheManager::disclaimAllCodeCaches()
+   {
+   if (!_disclaimEnabled)
+      return 0;
+
+   int32_t numDisclaimed = 0;
+
+#ifdef LINUX
+   TR::CompilationInfo *compInfo = TR::CompilationInfo::get(_jitConfig);
+   bool canDisclaimOnSwap = TR::Options::getCmdLineOptions()->getOption(TR_DisclaimMemoryOnSwap) && !compInfo->isSwapMemoryDisabled();
```

Comment:
if swapMemory is disabled and if the memory segment for the code cache is not backed by a file, the disclaim will fail.
We may need some sort of checking here.

Reply:
I think in that case` numDisclaimed ` will be zero. Are you thinking of some other checking?

Reply:
Right, added checks inside `J9::OptionsPostRestore::postProcessInternalCompilerOptions`. It would be much more work right now to check current status of code cache segments due to the current design.